### DISCUSSION TITLE
We were unable to remove a wazuh agent - Unable to connect to socket '/queue/db/wdb'.

### DISCUSSION THREAD

**Discussion by sidiyo** [wazuh/wazuh#16227] [Q&A]
Greetings, 

when trying to remove an agent from the manager, we received the message that we could not connect to socket '/queue/db/wdb'.

Available agents:
   ID: 001, Name: srvsostand, IP: 192.168.20.30
   ID: 007, Name: SRVAD02, IP: any
   ID: 013, Name: XXXXXXXXX, IP: any
Provide the ID of the agent to be removed (or '\q' to quit): 013
Confirm deleting it?(y/n): y
2023/02/16 13:52:27 manage_agents: INFO: Cannot connect to '/queue/db/wdb': Connection refused (111). Waiting 1 seconds to reconnect.
2023/02/16 13:52:28 manage_agents: INFO: Cannot connect to '/queue/db/wdb': Connection refused (111). Waiting 2 seconds to reconnect.
2023/02/16 13:52:30 manage_agents: INFO: Cannot connect to '/queue/db/wdb': Connection refused (111). Waiting 3 seconds to reconnect.
2023/02/16 13:52:33 manage_agents: ERROR: Unable to connect to socket '/queue/db/wdb'.
2023/02/16 13:52:33 manage_agents: ERROR: Unable to connect to socket '/queue/db/wdb'.
2023/02/16 13:52:33 manage_agents: ERROR: Could not remove the agent 013. Error: -2.
2023/02/16 10:54:52 wazuh-agentd: INFO: Requesting a key from server: 192.168.20.17
2023/02/16 10:54:52 wazuh-agentd: INFO: No authentication password provided
2023/02/16 10:54:52 wazuh-agentd: INFO: Using agent name as: srvforeman.superinterop.com.br
2023/02/16 10:54:52 wazuh-agentd: INFO: Waiting for server reply
2023/02/16 10:54:52 wazuh-agentd: ERROR: Duplicated agent name: srvforeman.superinterop.com.br (from manager)
2023/02/16 10:54:52 wazuh-agentd: ERROR: Unable to add agent. (from manager)
2023/02/16 10:55:02 wazuh-agentd: WARNING: (4101): Waiting for server reply (not started). Tried: '192.168.20.17'.
2023/02/16 10:55:02 wazuh-agentd: WARNING: Unable to connect to any server.
2023/02/16 10:55:02 wazuh-agentd: INFO: Closing connection to server (192.168.20.17:1514/tcp).
2023/02/16 10:55:02 wazuh-agentd: INFO: Trying to connect to server (192.168.20.17:1514/tcp).

Could someone give us an idea why? we tried changing the permissions of the ossec.conf file as we saw that it worked for one user, but it didn't work for us, because when we change the permissions, when starting wazuh it resets all the permissions.

Grateful so far.

**Comment by sidiyo** [comment]
Yesterday it was working normally, but today we are already having these problems.

We even noticed that wazuh is doing the discovery automatically, making it duplicate the servers.

**Reply by MiguelazoDS** [reply]
Hi @sidiyo, 

Please edit the `/var/ossec/etc/internal_options.conf` file with `wazuh_db.debug=2`. 

What is your Wazuh version? Restarting the manager does not help? What does the log say?

**Comment by sidiyo** [comment]
Thanks for the feedback,

I managed to solve it as follows, as we were trying to disable some detection rules, we ended up commenting a few more rules, when we uncommented the code, everything started working again.

**Reply by MiguelazoDS** [reply]
Glad you could make it work again!
